x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -236,6 +247,25 @@ public Realm realm(String name) {
         if (Strings.hasText(duplicateRealms)) {
             throw new IllegalArgumentException("Found multiple realms configured with the same name: " + duplicateRealms + "");
         }
+
+        if (missingOrderRealmSettingKeys.size() > 0) {
+            new DeprecationLogger(logger).deprecated("Found realms without order config: [{}]. " +
+                "In next major release, node will fail to start with missing realm order.",
+                String.join("; ", missingOrderRealmSettingKeys)
+            );
+        }
+        final List<String> duplicatedRealmOrderSettingKeys = orderToRealmOrderSettingKeys.entrySet()
+                .stream()
+                .filter(e -> e.getValue().size() > 1)
+                .map(e -> e.getKey() + ": " + String.join(",", e.getValue()))
+                .sorted()
+                .collect(Collectors.toList());
+        if (false == duplicatedRealmOrderSettingKeys.isEmpty()) {
+            new DeprecationLogger(logger).deprecated("Found multiple realms configured with the same order: [{}]. " +
```

Comment:
One comment if we keep this: Generally we create one `static` `DeprecationLogger` for each class similar to the regular logger. It's not too much overhead to create new ones but it's still creating a few unnecessary objects.

Reply:
Updated